registry.go

```diff
@@ -108,3 +134,80 @@ func (r *Registry) Retag(image, sha, tag string) (string, error) {
 	}
 	return retaggedNameWithTag, nil
 }
+
+func (r *Registry) getToken(req *http.Request) (string, error) {
+	const authorizationHeaderKey = "Www-Authenticate"
```

Comment:
Does the casing here matter? I did a quick google and saw some examples also using `WWW-Authenticate`.

Reply:
Very good point, on aliyun we have `Www-Authenticate` but even in the RFC they talk about `WWW-Authenticate` https://tools.ietf.org/html/rfc6750#section-3. I've changed the code to iterate through the header keys and check against a lowercased version, PTAL.

Reply:
Nice! Good that the comparison is now case insensitive. I can definitely imagine some other registry using a different case and that could be nasty to debug!